cfgov/unprocessed/apps/analytics-gtm/js/util/analytics-util.js

```diff
@@ -12,13 +28,61 @@ function addEventListenerToElem( elem, event, callback ) {
   }
 }
 
-const delay = ( function() {
+/**
+ * Log a message to the console if the `debug-gtm` URL parameter is set.
+ * @param {string} msg - Message to load to the console.
+ */
+function analyticsLog( ...msg ) {
+  if ( getQueryParameter( 'debug-gtm' ) === true ) {
+    console.log( `ANALYTICS DEBUG MODE: ${ msg }` );
+  }
+}
+
+/* Search for support of the matches() method by looking at
+   browser prefixes.
+   @param {HTMLNode} elem
+   The element to check for support of matches() method.
+   @returns {Function} The appropriate matches() method of elem. */
+function _getMatchesMethod( elem ) {
+  return elem.matches ||
+       elem.webkitMatchesSelector ||
+       elem.mozMatchesSelector ||
+       elem.msMatchesSelector;
+}
+
+/**
+ * Get the nearest parent node of an element.
+ *
+ * @param {HTMLNode} elem - A DOM element.
+ * @param {string} selector - CSS selector.
+ * @returns {HTMLNode} Nearest parent node that matches the selector.
+ */
+function closest( elem, selector ) {
```

Comment:
I think we have three [separate](https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/unprocessed/js/modules/util/dom-traverse.js#L66) versions of this now. We should probably consolidate them or just use an npm module.

Reply:
Yeah that makes sense, though I think an npm module would be the way to go. I don't think analytics should be reaching out into the rest of the codebase. Actually we should probably look into refactoring it away here in the analytics. If we had the architecture for components to talk to each other, analytics could just listen to what's happening in the components and wouldn't need to snoop around the dom like this.

Reply:
>> If we had the architecture for components to talk to each other, analytics could just listen to what's happening in the components and wouldn't need to snoop around the dom like this.

Yep, once I get done with the rate checker, it's up next. 

>> I think an npm module would be the way to go

Yeah, just pick one and I'll remove the other code to use that one as well.

Reply:
👍 I'll get it in a PR after this one. I think I could actually refactor it away.